src/socket.c

```diff
@@ -561,6 +581,10 @@ mrb_socket_sockaddr_family(mrb_state *mrb, mrb_value klass)
 static mrb_value
 mrb_socket_sockaddr_un(mrb_state *mrb, mrb_value klass)
 {
+#ifdef _WIN32
+  mrb_sys_fail(mrb, "sockaddr_un unsupported on Windows");
```

Comment:
`mrb_raise()` is better here.  `mrb_sys_fail()` is an utility function to raise an exception appropriate for the value of `errno`.  "sockaddr_un" error is not relevant to `errno`.